src/tests/utests/xi_utest_sft_logic.c

```diff
@@ -231,6 +231,292 @@ XI_TT_TESTCASE_WITH_SETUP(
         xi_sft_free_context( &sft_context );
     } )
 
+XI_TT_TESTCASE_WITH_SETUP( xi_utest__select_next_resource_to_download__null_context,
+                           xi_utest_setup_basic,
+                           xi_utest_teardown_basic,
+                           NULL,
+                           {
+                               xi_state_t state =
+                                   _xi_sft_select_next_resource_to_download( NULL );
+
+                               tt_want_int_op( XI_INVALID_PARAMETER, ==, state );
+                           } )
+
+XI_TT_TESTCASE_WITH_SETUP(
+    xi_utest__select_next_resource_to_download__picks_first_element,
+    xi_utest_setup_basic,
+    xi_utest_teardown_basic,
+    NULL,
+    {
+        xi_control_message_file_desc_ext_t three_file_list[3];
+        memset( three_file_list, 0, 3 * sizeof( xi_control_message_file_desc_ext_t ) );
```

Comment:
this one liner does the same as the two here:
`xi_control_message_file_desc_ext_t three_file_list[3] = {0};`
https://stackoverflow.com/questions/2589749/how-to-initialize-array-to-0-in-c

Reply:
cool cool.

Reply:
unfortunately the compiler does not like this. :(

Reply:
right, since the members aren't primitive type but struct, so ok, memset is the tool to zero it out 👍